server/src/main/java/com/witness/server/configuration/SecurityConfig.java

```diff
@@ -77,12 +79,16 @@ protected void configure(HttpSecurity http) throws Exception {
         .httpBasic().disable()
         .exceptionHandling()
         .authenticationEntryPoint(unauthorizedHandler()).and()
-        .authorizeRequests()
-        .antMatchers(restSecProps.getAllowedPublicApis().toArray(String[]::new)).permitAll()
-        .antMatchers(HttpMethod.OPTIONS, "/**").permitAll()
-        .anyRequest().fullyAuthenticated().and()
-        .addFilterBefore(tokenAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
+        .authorizeHttpRequests((authorize) -> authorize
```

Comment:
Although it's somewhat nicely documented, this part cost me almost 3 hours of debugging.
Starting with Spring Security 6.0 (which is included by Spring Boot 3.0), the `antMatchers` methods have been removed. The only remaining alternative, `requestMatchers`, resolves the type of matcher to use dynamically. In a MvcServlet project (which we have), it results in Spring MvcMatchers, otherwise AntMatchers (see https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#use-new-requestmatchers - although this dynamic resolution can only be found when debugging the framework source code..would have helped me a lot if it was noted there). That's mostly fine, but in some cases, the different behaviours of AntMatcher and MvcMatcher creates problems. I was not able to get our CORS and security filter configuration working such that both the `/h2-console` URL as well as `/swagger-ui` URIs were let through/allowed. It was very weird be - I only managed to make exactly one of them work, with different configs/notations in `application.yaml` (`SecurityProperties.java`).

In theory, I could have created a new yaml config property for public URLs/path specifications that _need_ AntMatchers (in favor of MvcMatchers). But I thought it to be too much of a hassle - that's why I simply gave up and force the use of AntMatchers for all of our public URLs. I couldn't find a deprecation hint (because Webflux still uses them as default) and nothing that said using AntMatchers would be discouraged. It's just that MvcMatcher, in theory, are more secure because they are a little stricter by default. But I think it's fine for us here.

Reply:
Two questions because I do not fully understand:

1. Can we now access `/h2-console` **and** `/swagger-ui` at the same time?
2. Do we use different matchers for different (types of?) URLs? If so, which URLs use which matcher?